### PR TITLE
TS, Reason, ReScript: Add implementation references

### PR DESCRIPTION
**Pull request by bragur**
Adding Implementation references for
- [x] Typescript
- [x] ReasonML
- [x] Rescript

Fixed type for Javascript reference

### CODE REVIEW

**Comment by hugihlynsson** [comment]
#### 📝Changed routes:


| [**…n/reference/typescript**](https://docs-9jb1v21hc.avo-dev.app/implementation/reference/typescript) |
|:-:|
|<a href="https://docs-9jb1v21hc.avo-dev.app/implementation/reference/typescript">Screenshot of /implementation/reference/typescript</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/hbG8WbcMivSrIG3rjM1eeBpB2qMJlZ9PzlaPjo2fVEegjmvIvJKWE2j_KFKfdNPHUNSTl42FIfllgLanu1_mCw.png">(view full size)</a> |







Commit 23fb6490d85be4903172fe12210b66f53a5e16a3 (https://docs-9jb1v21hc.avo-dev.app).

File: pages/implementation/reference/typescript.mdx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+Avo functions usage consists of 4 steps.
+
+[//]: # (TODO For future: Here we can add link to the "Type safe code & unit tests" docs section when that's ready, something like:)
+[//]: # 'TODO Learn more about how Avo Functions can help decrease time spent on implementing analytics and increase data reliability in this guide.'
+
+#### Step 1. Include the Avo file
+
+##### Pull the generated code with the Avo CLI
+
+To get the Avo generated Typescript file you must be a member of an Avo workspace with a JavaScript source. Ask for an invite from a colleague or create a <a href="https://avo.app/onboarding" target="_blank">new workspace</a>
```
Comment (alex-tpom6oh) [inline pages/implementation/reference/typescript.mdx:27]: JavaScript -> TypeScript here

File: pages/implementation/reference/typescript.mdx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+#### Step 4. Verify the implementation
+
+Use the <Link passHref href="/workspace/tracking-plan/implementation-status"><a>Implementation status</a></Link> in your Avo workspace and the <Link passHref href="/explore-tracking-plan/start-using-visual-debuggers"><a>visual debuggers</a></Link> to verify that your implementation is correct.
+
+### Reference
+
+#### initAvo
+
+```typescript
+Avo.initAvo(
```
Comment (alex-tpom6oh) [inline pages/implementation/reference/typescript.mdx:90]: In the recent docs I went with stating actual method signatures here, like
```
export function initAvo(options: {env: AvoEnv; [webDebugger?: boolean];
  [strict?: boolean]; [noop?: boolean];
  [reportFailureAs?: 'error' | 'warn' | 'log']; [inspector?: AvoInspector];
  [avoLogger?: AvoLogger]},
  [systemProperties: {
    systemProperty0: number;
    systemProperty1: boolean;
    ...}],
  [destinationOptions: any], [CustomDestDestination: CustomDestination]) ```
  
  I think it's more clear in the statically typed languages 🤔

File: pages/implementation/reference/typescript.mdx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+###### Additional arguments
+
+`anonymousId_`: string, Node.js only, this argument is automatically added if corresponding setting is enabled, used to track anonymous users
+
+`segmentContext_`: object, Node.js only, passed down to Segment as the Segment context, e.g. `segment.track({..., context: context})`
+
+### Destinations
+
+You can send your data using the Avo generated JavaScript code to any data destination that accepts custom events, including:
```
Comment (alex-tpom6oh) [inline pages/implementation/reference/typescript.mdx:197]: JavaScript here

File: pages/implementation/reference/reasonml.mdx [hunk 4]
```diff
@@ -104,7 +104,7 @@ Initializes Avo, needs to be called before the tracking methods. This method wil
 
 - `env`: AvoEnv, can be set to dev, prod and staging.
 - `systemProperties`: AvoSystemProperties.t, where each field represents a system property in your tracking plan.
-- `webDegubber`: 
+- `webDebugger`:
```
Comment (alex-tpom6oh) [inline pages/implementation/reference/reasonml.mdx:107]: hah, funny

File: pages/implementation/reference/rescript.mdx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+Avo functions usage consists of 4 steps.
+
+[//]: # (TODO For future: Here we can add link to the "Type safe code & unit tests" docs section when that's ready, something like:)
+[//]: # 'TODO Learn more about how Avo Functions can help decrease time spent on implementing analytics and increase data reliability in this guide.'
+
+#### Step 1. Include the Avo file
+
+##### Pull the generated code with the Avo CLI
+
+To get the Avo generated Rescript file you must be a member of an Avo workspace with a Rescript source. Ask for an invite from a colleague or create a <a href="https://avo.app/onboarding" target="_blank">new workspace</a>
```
Comment (alex-tpom6oh) [inline pages/implementation/reference/rescript.mdx:27]: Should we use capital s in ReScript?

Comment (bragur) [reply pages/implementation/reference/rescript.mdx]: Of course! ☺️

File: pages/implementation/reference/typescript.mdx [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+Initializes Avo, needs to be called before the tracking methods. This method will initialize the SDKs for all destinations that are managed by Avo, and call the `make(env)` callback in the provided custom destinations.
+
+##### Arguments
+
+`options`: `{env, [noop], [strict], [avoLogger], [inspector], [mobileDebugger]}`
+
+- `env`: AvoEnv, can be set to dev, prod and staging.
+- `[webDebugger]: optional bool, for optional Avo Web Debugger instance
+- `[strict = false]`: optional bool, if set, Avo will throw an exception when it detects a tracking problem in development or staging. Note that the strict flag is ignored in production.
+- `[noop = false]`: optional bool, if set, Avo won't make any network calls (no tracking) in development and staging environments. Note that the noop flag is ignored in production.
```
Comment (alex-tpom6oh) [inline pages/implementation/reference/typescript.mdx:117]: I think optional is not the correct term here, probably better to call it bool with a default value. Could you also fix it in JS while you are on it?

Comment (alex-tpom6oh) [reply pages/implementation/reference/typescript.mdx:117]: same for strict

Comment (bragur) [reply pages/implementation/reference/typescript.mdx]: Like this?

Comment (alex-tpom6oh) [reply pages/implementation/reference/typescript.mdx]: Yes, looks very accurate to me now

**Review by alex-tpom6oh — APPROVED** [review]

**Pull request merged**